packages/legend-application-query/src/stores/QueryBuilderTypeAheadSearchHelper.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  type AbstractPropertyExpression,
+  type ExecutionResult,
+  type ValueSpecification,
+  PrimitiveInstanceValue,
+  PRIMITIVE_TYPE,
+  TdsExecutionResult,
+} from '@finos/legend-graph';
+import { guaranteeType, isNonNullable, isString } from '@finos/legend-shared';
+import { QueryBuilderPostFilterOperator_StartWith } from './postFilterOperators/QueryBuilderPostFilterOperator_StartWith.js';
+import type { QueryBuilderAggregateColumnState } from './QueryBuilderAggregationState.js';
+import {
+  PostFilterConditionState,
+  QueryBuilderPostFilterTreeConditionNodeData,
+} from './QueryBuilderPostFilterState.js';
+import {
+  QueryBuilderProjectionColumnState,
+  QueryBuilderSimpleProjectionColumnState,
+} from './QueryBuilderProjectionState.js';
+import { QueryBuilderState } from './QueryBuilderState.js';
+
+const SEARCH_TAKE = 200;
+const START_LENGTH = 2;
+
+const createAndSetupQueryBuilderStateForTypeAhead = (
```

Comment:
Nice idea here to reuse the `query builder state` to quickly build a query, but I think we should not since we tie this to the implementation of query builder state, while it's mainly a convenience thing; in reality, we kind of want to follow what we do for `preview data` which is to construct a query from scratch. Now arguably, that's painful, but there are a few tools in the arsenal for us to take advantage of:

1. The lambda builder utilities
2. Now that we have properly extended `AbstractPureGraphManager`, feel free to have a template query in `V1_ValueSpecification` JSON format and just chuck in the relevant part.